src/os/bluestore/KernelDevice.cc

```diff
@@ -256,6 +256,15 @@ void KernelDevice::_aio_thread()
 		 << " ioc " << ioc
 		 << " with " << left << " aios left" << dendl;
 	assert(r >= 0);
+
+	std::map<uint64_t, bufferlist*>::iterator it =  ioc->aio_read_queue.find(aio[i]->offset);
+	if (it != ioc->aio_read_queue.end()) {
+	  bufferlist *bl = it->second;
+	  bl->clear();
+	  bl->claim_append(aio[i]->bl);
+	  --ioc->num_reading;
+	}
+
```

Comment:
At firstly, i used this method which check read/write type. But 
In aio_write:
>>if (g_conf->bdev_inject_crash &&
>>	rand() % g_conf->bdev_inject_crash == 0) {
>>      derr << __func__ << " bdev_inject_crash: dropping io " << off << "~" << len
>>	   << dendl;
>>      // generate a real io so that aio_wait behaves properly, but make it
>>      // a read instead of write, and toss the result.
>>      aio.pread(off, len);
>>      ++injecting_crash;
In fact,  we can get io type from 'struct iocb".(see: http://compbio.cs.toronto.edu/repos/snowflock/xen-3.0.3/tools/libaio/src/libaio.h)
We can firstly get the type and then still need check aio_read_queue.